libsnark/zk_proof_systems/plonk/common_input.tcc

```diff
@@ -0,0 +1,138 @@
+    plonk_compute_public_input_polynomial(
+        PI_points, this->PI_poly, this->L_basis);
+    //    plonk_interpolate_over_lagrange_basis<Field>(PI_points, this->PI_poly,
+    //    this->L_basis);
+#ifdef DEBUG
+    printf("[%s:%d] this->PI_poly\n", __FILE__, __LINE__);
+    print_vector(this->PI_poly);
+#endif // #ifdef DEBUG
+
+    // compute the selector polynomials (q-polynomials) from the
+    // transposed gates matrix over the Lagrange basis q_poly = \sum_i
+    // q[i] * L[i] where q[i] is a coefficient (a scalar Field
+    // element) and L[i] is a polynomial with Field coefficients
+    this->Q_polys.resize(this->num_qpolys, polynomial<Field>(this->num_gates));
+    plonk_compute_selector_polynomials<Field>(
+        gates_matrix_transpose, this->Q_polys, this->L_basis);
+#ifdef DEBUG
+    for (int i = 0; i < (int)this->num_qpolys; ++i) {
+        printf("\n[%s:%d] this->Q_polys[%2d]\n", __FILE__, __LINE__, i);
+        print_vector(this->Q_polys[i]);
+    }
+#endif // #ifdef DEBUG
+
+    // number of generators for H, Hk1, Hk2
+    int num_hgen = NUM_HGEN;
+    // omega[0] are the n roots of unity, omega[1] are omega[0]*k1,
+    // omega[2] are omega[0]*k2
+    //    std::vector<std::vector<Field>> omega;
+    this->omega_roots.resize(num_hgen, std::vector<Field>(this->num_gates));
+    plonk_compute_roots_of_unity_omega(k1, k2, this->omega_roots);
+    // H_gen contains the generators of H, k1 H and k2 H in one place
+    // ie. this->omega_roots, this->omega_roots_k1 and this->omega_roots_k2
+    plonk_roots_of_unity_omega_to_subgroup_H(this->omega_roots, this->H_gen);
+#ifdef DEBUG
+    printf("[%s:%d] this->H_gen\n", __FILE__, __LINE__);
+    print_vector(this->H_gen);
+    for (int i = 0; i < (int)this->H_gen.size(); ++i) {
+        assert(this->H_gen[i] == example.H_gen[i]);
+    }
+#endif // #ifdef DEBUG
+
+    // permute this->H_gen according to the wire permutation
+    this->H_gen_permute.resize(num_hgen * this->num_gates, Field(0));
+    plonk_permute_subgroup_H<Field>(
+        this->H_gen_permute, this->H_gen, wire_permutation);
+#ifdef DEBUG
+    printf("[%s:%d] this->H_gen_permute\n", __FILE__, __LINE__);
+    print_vector(this->H_gen_permute);
+    for (size_t i = 0; i < this->H_gen_permute.size(); ++i) {
+        assert(this->H_gen_permute[i] == example.H_gen_permute[i]);
+    }
+#endif // #ifdef DEBUG
+
+    // compute the permutation polynomials S_sigma_1, S_sigma_2,
+    // S_sigma_3 (see [GWC19], Sect. 8.1) (our indexing starts from 0)
+    this->S_polys.resize(num_hgen, polynomial<Field>(this->num_gates));
+    plonk_compute_permutation_polynomials<Field>(
+        this->S_polys, this->H_gen_permute, this->L_basis, this->num_gates);
+#ifdef DEBUG
+    for (int i = 0; i < num_hgen; ++i) {
+        printf("[%s:%d] this->S_polys[%d]\n", __FILE__, __LINE__, i);
+        print_vector(this->S_polys[i]);
+    }
+#endif // #ifdef DEBUG
+}
+} // namespace libsnark
+
+#endif // PLONK_PPZKSNARK_COMMON_INPUT_TCC_
```

Comment:
FYI we usually use the file path for the include guards, see: https://github.com/clearmatics/libsnark/blob/develop/libsnark/gadgetlib1/gadgets/curves/scalar_multiplication.tcc#L826 for instance. So here we'd have `LIBSNARK_ZK_PROOF_SYSTEMS_PLONK_COMMON_INPUT_TCC_`. Note, this convention isn't used consistently across the libsnark project, (e.g. https://github.com/clearmatics/libsnark/blob/develop/libsnark/zk_proof_systems/ppzksnark/r1cs_gg_ppzksnark/r1cs_gg_ppzksnark.hpp#L47). This is in part due to the fact that the original authors and us have picked different coding conventions and we update the code base organically, as we go.

In any case your include guard isn't right here since `PPZKSNARK` isn't in the path for this file. In fact, you created a specific `plonk` folder right under `zk_proof_systems` for the plonk code. The "ppzksnarks" (i.e. Pre-Processing ZKSNARKs) are all under `zk_proof_systems/ppzksnark`

Reply:
Thanks, @AntoineRondelet ! Naming of all include guards corrected.

As a side note: form what you explain, wouldn't it be more appropriate to move the plonk folder under zk_proof_systems/ppzksnark then?